pages/api/augment-request.ts

```diff
@@ -38,148 +78,162 @@ function castOracleNameForOOUi(oracleType: string): string {
     case "SkinnyOptimisticOracle":
       return "Skinny";
     default:
-      return "";
+      throw new Error("Unable to cast oracle name for OO UI: " + oracleType);
   }
 }
 
-async function constructOptimisticOraclesByChain(chainId: SupportedChainIds) {
-  const optimisticOracle = (await constructContract(
-    chainId,
-    "OptimisticOracle"
-  )) as OptimisticOracleEthers;
-  const optimisticOracleV2 = (await constructContract(
-    chainId,
-    "OptimisticOracleV2"
-  )) as OptimisticOracleV2Ethers;
-
-  // Only mainnet has a SkinnyOptimisticOracle so only construct it here. All other chains of interest have OOv1 and V2.
-  const skinnyOptimisticOracle =
-    chainId == 1
-      ? ((await constructContract(
-          chainId,
-          "SkinnyOptimisticOracle"
-        )) as SkinnyOptimisticOracleEthers)
-      : null;
-  return [optimisticOracle, optimisticOracleV2, skinnyOptimisticOracle];
+type LookupTable<E> = Record<string, Record<number, E>>;
+function createLookupTable<Event extends { time: number; identifier: string }>(
+  events: Event[],
+  table: LookupTable<Event> = {}
+): LookupTable<Event> {
+  return events.reduce((table, event) => {
+    const identifier = event.identifier.toLowerCase();
+    const time = event.time;
+    if (identifier && table[identifier] === undefined) table[identifier] = {};
+    if (time && table[identifier][time] === undefined)
+      table[identifier][time] = event;
+    return table;
+  }, table);
 }
 
-async function getOptimisticOraclePriceRequestEventsByChainId(
+async function getVotingPriceRequestAdded(
+  contractType: VotingType,
   chainId: SupportedChainIds
-) {
-  const chainOptimisticOracles = await constructOptimisticOraclesByChain(
-    chainId
+): Promise<CommonEventData[]> {
+  const contract = await constructContract(chainId, contractType);
+  const events = await contract.queryFilter(
+    contract.filters.PriceRequestAdded(),
+    getFromBlock(contractType as string, chainId as number)
```

Comment:
does changing the `fromBlock` actually modify this if the difference is 0 or the deployment block? so say that a contract is deployed at block 100. does it actually save request size and time to set the value in a `queryFilter` to set the value to 100, as apposed to 0? I thought that it just shortcut given how the event lookup works in infura.

Reply:
i didnt know how intelligent event queries were, i assumed the worst, it would be great if he started from contract deployment automatically. would make this feature redundant, but technically you could bump this configuration after the deploy time if you didn't care about data over x days old.  i havent seen any deterioration in behavior since this change though, so I dont think its any worse than before.